src/colorize.cr

```diff
@@ -123,43 +123,63 @@ class Object
   include Colorize::ObjectExtensions
 end
 
-struct Colorize::Object(T)
-  private FORE_DEFAULT       = "39"
-  private FORE_BLACK         = "30"
-  private FORE_RED           = "31"
-  private FORE_GREEN         = "32"
-  private FORE_YELLOW        = "33"
-  private FORE_BLUE          = "34"
-  private FORE_MAGENTA       = "35"
-  private FORE_CYAN          = "36"
-  private FORE_LIGHT_GRAY    = "37"
-  private FORE_DARK_GRAY     = "90"
-  private FORE_LIGHT_RED     = "91"
-  private FORE_LIGHT_GREEN   = "92"
-  private FORE_LIGHT_YELLOW  = "93"
-  private FORE_LIGHT_BLUE    = "94"
-  private FORE_LIGHT_MAGENTA = "95"
-  private FORE_LIGHT_CYAN    = "96"
-  private FORE_WHITE         = "97"
-
-  private BACK_DEFAULT       = "49"
-  private BACK_BLACK         = "40"
-  private BACK_RED           = "41"
-  private BACK_GREEN         = "42"
-  private BACK_YELLOW        = "43"
-  private BACK_BLUE          = "44"
-  private BACK_MAGENTA       = "45"
-  private BACK_CYAN          = "46"
-  private BACK_LIGHT_GRAY    = "47"
-  private BACK_DARK_GRAY     = "100"
-  private BACK_LIGHT_RED     = "101"
-  private BACK_LIGHT_GREEN   = "102"
-  private BACK_LIGHT_YELLOW  = "103"
-  private BACK_LIGHT_BLUE    = "104"
-  private BACK_LIGHT_MAGENTA = "105"
-  private BACK_LIGHT_CYAN    = "106"
-  private BACK_WHITE         = "107"
+module Colorize
+  alias Color = ColorANSI | Color256 | ColorRGB
+
+  enum ColorANSI
+    Default      = 39
```

Comment:
Please, use `String`s like before, to avoid creating intermediary allocations in `#back` and `#fore`.

Reply:
I am considering to rewrite `ColorANSI` definition as:

```crystal
struct ColorANSI
  private def initialize(@fore : String, @back)
  end

  def fore(io : IO)
    @fore.to_s io
  end

  def back(io : IO)
    @back.to_s io
  end

  Default = new("30", "40")
  # ...
end
```

What do you think?

/cc @ysbaddaden

Reply:
I like the Enum, it's simpler.

The real benefit is writing directly to the IO in the `#back` and `#fore` methods of `ColorANSI`, `Color256` and `ColorRGB` to avoid intermediary String allocations.